src/app/advanced-simulator/simulator-modal/simulator-modal.component.ts

```diff
@@ -0,0 +1,141 @@
+import { Component, inject } from '@angular/core';
+import { AlertService, gettext, OptionsService } from '@c8y/ngx-components';
+import { BsModalRef } from 'ngx-bootstrap/modal';
+import { PROMPT } from './prompt';
+import { FetchClient } from '@c8y/client';
+import Anthropic from '@anthropic-ai/sdk';
+import { Router } from '@angular/router';
+
+const AI_MODEL = 'claude-3-5-sonnet-20240620';
+
+@Component({
+  selector: 'c8y-simulator-modal',
+  templateUrl: './simulator-modal.component.html',
+})
+export class SimulatorModalComponent {
+  title = gettext('Add advanced simulator');
+  useCase = '';
+  appOptions = inject(OptionsService);
+  apiKey = this.appOptions.get('apiKey');
+  result: Promise<any> = new Promise((resolve, reject) => {
+    this._close = resolve;
+    this._reject = reject;
+  });
+  pending = false;
+  instances = 1;
+  details = '';
+  private _close: ((_) => void) | undefined;
+  private _reject: (reason?: any) => void;
+
+  private modalRef = inject(BsModalRef);
+  private client = inject(FetchClient);
+  private alertService = inject(AlertService);
+  private router = inject(Router);
+
+  cancel() {
+    this._reject();
+    this.modalRef.hide();
+  }
+
+  async create() {
+    try {
```

Comment:
we were talking about adding api key to session storage... it could be done when first simulator is created. now I have to paste api key every time